src/select/parts/item.tsx

```diff
@@ -50,6 +53,14 @@ const Item = (
   const isChild = option.type === 'child';
   const wrappedOption: OptionDefinition = option.option;
   const disabled = option.disabled || wrappedOption.disabled;
+  const disabledReason =
+    disabled && (option.disabledReason || wrappedOption.disabledReason)
+      ? option.disabledReason || wrappedOption.disabledReason
+      : '';
```

Comment:
Why do you need to check both `option` and `wrappedOption`?

Reply:
Good catch. Only `wrappedOption` could be checked. Updated